app/javascript/src/RSCMarkdownPage/ror_components/RSCMarkdownPage.jsx

```
@@ -0,0 +1,344 @@
+  const markdown = initialText || defaultMarkdown;
+
+  // Server-side markdown processing using 'marked'
+  // This heavy library stays on the server and never ships to client
+  const processedHtml = await marked(markdown, {
+    gfm: true, // GitHub Flavored Markdown
+    breaks: true,
+  });
+
+  return (
+    <>
+      <style dangerouslySetInnerHTML={{ __html: `
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)